include/tvm/relay/pass_manager.h

```diff
@@ -0,0 +1,393 @@
+/*!  * Copyright (c) 2019 by Contributors
+ * \file tvm/relay/pass_manager.h
+ *
+ * \brief The pass manager manages a sequence of Relay-to-Relay transformation
+ * passes over a particlar unit of AST. The design is largely inspired from
+ * LLVM's pass manager.
+ *
+ * The responsibilities of a pass manager usually involves:
+ *  - Organizing the execution order of optimization passes though not
+ * necessarily in the optimal sequence.
+ *  - Collecting required analysis information and keep them up-to-date.
+ *  - Reducing the effort required to implement new passes for compiler
+ * developers, etc.
+ *
+ * TODO(jroesch, zhiics): We are currently using a very simple design for the
+ * pass manager, i.e. it executes a specific pass or sequence of passes.
+ *
+ * In the future we need to describe constraints between passes. For example,
+ * we may want to preserve dependencies between different passes and validate
+ * them on the completion of a certain pass.
+ *
+ * We also need to store side information and import the error reporting system.
+ */
+#ifndef TVM_RELAY_PASS_MANAGER_H_
+#define TVM_RELAY_PASS_MANAGER_H_
+
+#include <tvm/attrs.h>
+#include <tvm/ir.h>
+#include <tvm/packed_func_ext.h>
+#include <tvm/relay/error.h>
+#include <tvm/relay/expr.h>
+#include <tvm/relay/module.h>
+#include <tvm/relay/type.h>
+
+#include <string>
+#include <vector>
+
+namespace tvm {
+namespace relay {
```

Comment:
Let us consider directly merge this into relay/pass.h

Reply:
@tqchen Thanks. I considered this, but I think it is probably necessary to keep it in the pass_manager.h file for transitioning purpose.